Objects/unicodeobject.c

```diff
@@ -4700,6 +4700,8 @@ static Py_ssize_t
 ascii_decode(const char *start, const char *end, Py_UCS1 *dest)
 {
     const char *p = start;
+    const char *size_t_end  = end - SIZEOF_SIZE_T;
+    const char *unrolled_end = end - (4 * SIZEOF_SIZE_T);
```

Comment:
@encukou Should we still use macros such as `SIZEOF_SIZE_T` and `ALIGNOF_SIZE_T`, or should we prefer `sizeof` and [`_Alignof`](https://en.cppreference.com/w/c/language/_Alignof)?

Reply:
IMO, we should prefer `sizeof` and `_Alignof` *here*.
Things are a bit different in public headers (where `sizeof` is OK but `_Alignof` isn't) or the preprocessor (in `#if`, neither work).

(There is no C API WG formal guideline yet; this is a personal recommendation. If anyone wants a wider discussion, do that in [Discourse](https://discuss.python.org/c/core-dev/c-api/30).)